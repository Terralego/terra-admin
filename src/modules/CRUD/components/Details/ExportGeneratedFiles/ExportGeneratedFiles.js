import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { Classes, MenuDivider, MenuItem } from '@blueprintjs/core';

const ExportGeneratedFiles = ({ files }) => {
  const { t } = useTranslation();

  const count = files.length;

  if (!count) {
    return null;
  }

  return (
    <>
      <MenuDivider title={t('CRUD.details.generatedDocument', { count })} />
      {files.map(({ download_url: url, template_file: file, template_name: name }) => (
        <MenuItem
          className={Classes.MINIMAL}
          icon="document"
          text={name}
          key={file}
          href={url}
        />
      ))}
    </>
  );
};

ExportGeneratedFiles.propTypes = {
  files: PropTypes.arrayOf(
    PropTypes.shape({
      template_name: PropTypes.string,
      download_url: PropTypes.string,
      template_file: PropTypes.string,
    }),
  ),
};

ExportGeneratedFiles.defaultProps = {
  files: [],
};

export default ExportGeneratedFiles;
