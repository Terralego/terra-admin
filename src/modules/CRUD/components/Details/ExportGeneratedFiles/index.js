import ExportGeneratedFiles from './ExportGeneratedFiles';

export default ExportGeneratedFiles;
