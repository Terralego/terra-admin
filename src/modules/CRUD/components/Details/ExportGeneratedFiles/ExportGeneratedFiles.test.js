import React from 'react';
import renderer from 'react-test-renderer';

import ExportGeneratedFiles from './ExportGeneratedFiles';

jest.mock('@blueprintjs/core', () => ({
  MenuDivider: props => <li {...props} />,
  MenuItem: props => <li {...props} />,
  Classes: {
    MINIMAL: 'MINIMAL',
  },
}));


const props = {
  files: [{
    template_name: 'FileName',
    download_url: 'path/to/file/',
    template_file: 'file1.pdf',
  }, {
    template_name: 'FileName2',
    download_url: 'path/to/file2/',
    template_file: 'file2.pdf',
  }],
};


it('should render correctly', () => {
  const tree = renderer.create((
    <ExportGeneratedFiles
      {...props}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
