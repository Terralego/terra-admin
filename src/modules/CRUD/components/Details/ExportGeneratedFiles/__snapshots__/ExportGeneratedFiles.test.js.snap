// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <li
    title="CRUD.details.generatedDocument"
  />,
  <li
    className="MINIMAL"
    href="path/to/file/"
    icon="document"
    text="FileName"
  />,
  <li
    className="MINIMAL"
    href="path/to/file2/"
    icon="document"
    text="FileName2"
  />,
]
`;
